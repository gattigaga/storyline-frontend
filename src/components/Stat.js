import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { separateByComma } from "../helpers/formatter";

const Label = styled.p`
  font-family: Roboto Slab;
  font-size: 18px;
  color: #e74c3c;
  margin: 0px;
  text-align: center;
`;

const Value = styled.p`
  font-family: Roboto;
  font-size: 32px;
  color: #333;
  margin: 0px;
  text-align: center;
`;

const Stat = ({ label, value }) => (
  <div>
    <Value>{separateByComma(value)}</Value>
    <Label>{label}</Label>
  </div>
);

Stat.propTypes = {
  label: PropTypes.string,
  value: PropTypes.number
};

Stat.defaultProps = {
  value: 0
};

export default Stat;
