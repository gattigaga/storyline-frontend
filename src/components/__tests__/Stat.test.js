import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Stat from "../Stat";

describe("Stat", () => {
  const setup = propOverrides => {
    const props = {
      label: "Followers",
      value: 32000,
      ...propOverrides
    };

    const wrapper = shallow(<Stat {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
