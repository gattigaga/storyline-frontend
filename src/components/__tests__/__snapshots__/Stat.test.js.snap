// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stat should renders default 1`] = `
<div>
  <styled.p>
    32,000
  </styled.p>
  <styled.p>
    Followers
  </styled.p>
</div>
`;
